flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java

```diff
@@ -325,7 +328,7 @@ public void testSpecificSnapshotTimestamp() throws Exception {
     long snapshot0Timestamp = tableResource.table().currentSnapshot().timestampMillis();
 
     // sleep for 2 ms to make sure snapshot1 has a higher timestamp value
-    Thread.sleep(2);
+    Awaitility.await().atLeast(Duration.ofMillis(2)).until(() -> true);
```

Comment:
If thread just requires to sleep for some duration , ideally it should be like 
`Awaitility.await().pollDelay(2, TimeUnit.MILLISECONDS).until(() -> System.currentTimeMillis()-snapshot0Timestamp>2);`